src/aeso_fcode_to_fate.erl

```diff
@@ -886,6 +929,9 @@ attributes(I) ->
         loop                                  -> Impure(pc, []);
         switch_body                           -> Pure(none, []);
         'RETURN'                              -> Impure(pc, []);
+        {'DBG_LOC', _, _}                     -> Pure(none, []);
+        {'DBG_DEF', _, _}                     -> Pure(none, []);
+        {'DBG_UNDEF', _, _}                   -> Pure(none, []);
```

Comment:
`Pure` is used to mark instructions that don't alter the chain, so they can be moved around more freely. I think debug instructions should be considered `Impure`, so they are protected from compiler mangling.

`none` is fine, it means that the instruction does not write to anywhere nor alter our position

Reply:
Changed to `Impure` here c721eccf11fed7bca418f411a249e849841db7be